src/rtext.c

```diff
@@ -1344,40 +1344,49 @@ Rectangle GetGlyphAtlasRec(Font font, int codepoint)
 // Get text length in bytes, check for \0 character
 unsigned int TextLength(const char *text)
 {
-    unsigned int length = 0; //strlen(text)
-
-    if (text != NULL)
-    {
-        while (*text++) length++;
-    }
+    if (!text)
+        return 0;
 
+    // use strlen since it uses vector operations
+    unsigned int length = strlen(text);
+    
     return length;
 }
 
 // Formatting of text with variables to 'embed'
-// WARNING: String returned will expire after this function is called MAX_TEXTFORMAT_BUFFERS times
 const char *TextFormat(const char *text, ...)
 {
-#ifndef MAX_TEXTFORMAT_BUFFERS
-    #define MAX_TEXTFORMAT_BUFFERS 4        // Maximum number of static buffers for text formatting
-#endif
+    if (!text)
+        return "";
 
-    // We create an array of buffers so strings don't expire until MAX_TEXTFORMAT_BUFFERS invocations
-    static char buffers[MAX_TEXTFORMAT_BUFFERS][MAX_TEXT_BUFFER_LENGTH] = { 0 };
-    static int index = 0;
-
-    char *currentBuffer = buffers[index];
-    memset(currentBuffer, 0, MAX_TEXT_BUFFER_LENGTH);   // Clear buffer before using
+    static char buffer[MAX_TEXT_BUFFER_LENGTH] = { 0 };
+    memset(buffer, 0, MAX_TEXT_BUFFER_LENGTH);
 
     va_list args;
     va_start(args, text);
-    vsnprintf(currentBuffer, MAX_TEXT_BUFFER_LENGTH, text, args);
-    va_end(args);
 
-    index += 1;     // Move to next buffer for next function call
-    if (index >= MAX_TEXTFORMAT_BUFFERS) index = 0;
+    int length = vsnprintf(NULL, 0, text, args); // Determine the length of the formatted string
+    if (length < 0)
+    {
+        // Handle error
+        va_end(args);
+        return NULL;
+    }
+
+    if (buffer == NULL)
+    {
+        // Handle memory allocation failure
+        va_end(args);
+        return NULL;
+    }
+
+    va_end(args); // Reset the va_list
+
+    va_start(args, text);                      // Start again for the actual formatting
+    vsnprintf(buffer, length + 1, text, args); // Format the string and copy it to the result buffer
+    va_end(args);
 
-    return currentBuffer;
+    return buffer;
```

Comment:
Just as a reference (since I had this test ready), this `TextFormat()` change also looks like a considerable performance improvement for `PLATFORM_DESKTOP`. Test example:
```
#include <stdio.h>
#include <time.h>
#include "raylib.h"

int main(void) {

    {
        double timeRes = 0.0f;
        clock_t timeStart = clock();
        const char *testRes = TextFormat("%i", 11);
        clock_t timeEnd = clock();
        timeRes = (double)(timeEnd - timeStart) / CLOCKS_PER_SEC;
        printf("textFormat:  %f sec (%s)\n", timeRes, testRes);
    }

    {
        double timeRes = 0.0f;
        clock_t timeStart = clock();
        const char *testRes = TextFormat2("%i", 11);
        clock_t timeEnd = clock();
        timeRes = (double)(timeEnd - timeStart) / CLOCKS_PER_SEC;
        printf("textFormat2: %f sec (%s)\n", timeRes, testRes);
    }

    return 0;
}
```
Results on Linux Mint 21.1 64-bit:
```
~/test$ ./test
textFormat:  0.000070 sec (11)
textFormat2: 0.000006 sec (11)
~/test$ ./test 
textFormat:  0.000059 sec (11)
textFormat2: 0.000006 sec (11)
~/test$ ./test
textFormat:  0.000068 sec (11)
textFormat2: 0.000006 sec (11)

```
There were no performance gains on `PLATFORM_WEB`. Tested on Firefox 115.1.0esr 64-bit and Chromium 115.0.5790.170 64-bit both running on Linux Mint 21.1 64-bit.